src/call-api.ts

```diff
@@ -0,0 +1,152 @@
+import * as abi from "@api3/airnode-abi";
+import * as adapter from "@api3/airnode-adapter";
+import * as node from "@api3/airnode-node";
+import * as ois from "@api3/airnode-ois";
+import * as ethers from "ethers";
+import isNil from "lodash/isNil";
+import { ApiValuesByBeaconId, RrpBeaconServerKeeperTrigger } from "./types";
+import { retryGo } from "./utils";
+
+export const readApiValue = async (
+  airnodeHDNode: ethers.utils.HDNode,
+  oises: ois.OIS[],
+  apiCredentials: node.ApiCredentials[],
+  {
+    oisTitle,
+    endpointName,
+    templateId,
+    templateParameters,
+    overrideParameters,
+  }: RrpBeaconServerKeeperTrigger
+): Promise<node.LogsData<ApiValuesByBeaconId>> => {
+  const configParameters = [...templateParameters, ...overrideParameters];
+
+  // Derive beaconId
+  const encodedParameters = abi.encode(configParameters);
+  const beaconId = ethers.utils.solidityKeccak256(
+    ["bytes32", "bytes"],
+    [templateId, encodedParameters]
+  );
+
+  // Verify templateId matches data in rrpBeaconServerKeeperJob
+  const airnodeAddress = airnodeHDNode.derivePath(
+    ethers.utils.defaultPath
+  ).address;
+  const endpointId = ethers.utils.keccak256(
+    ethers.utils.defaultAbiCoder.encode(
+      ["string", "string"],
+      [oisTitle, endpointName]
+    )
+  );
+  const encodedTemplateParameters = abi.encode(templateParameters);
+  const expectedTemplateId = node.evm.templates.getExpectedTemplateId({
+    airnodeAddress,
+    endpointId,
+    encodedParameters: encodedTemplateParameters,
+    id: templateId,
+  });
+  if (expectedTemplateId !== templateId) {
+    return [
+      [
+        node.logger.pend(
+          "ERROR",
+          `templateId '${templateId}' does not match expected templateId '${expectedTemplateId}'`
+        ),
+      ],
+      { [beaconId]: null },
+    ];
+  }
+
+  const configOis = oises.find((o) => o.title === oisTitle)!;
+  const configEndpoint = configOis.endpoints.find(
+    (e) => e.name === endpointName
+  )!;
+  const apiCallParameters = configParameters.reduce(
+    (acc, p) => ({ ...acc, [p.name]: p.value }),
+    {}
+  );
+  const reservedParameters =
+    node.adapters.http.parameters.getReservedParameters(
+      configEndpoint,
+      apiCallParameters || {}
+    );
+  if (!reservedParameters._type) {
+    return [
+      [
+        node.logger.pend(
+          "ERROR",
+          `reserved parameter 'type' is missing for endpoint: ${endpointName}`
```

Comment:
```suggestion
          `reserved parameter '_type' is missing for endpoint: ${endpointName}`
```

Reply:
Nit: maybe we could save quite a bit of space in a few places with something like:

```ts
const message = `reserved parameter '_type' is missing for endpoint: ${endpointName}`;
const log = node.logger.pend('ERROR', message);
return [[log], { [beaconId]: null }];
```

3 lines vs 8 lines